pkgs/games/prismlauncher/wrapper.nix

```diff
@@ -37,28 +59,38 @@ symlinkJoin {
 
   nativeBuildInputs = [
     wrapQtAppsHook
-  ];
+  ]
+  ++ lib.optional shellWrapper makeWrapper;
 
   buildInputs = [
     qtbase
     qtsvg
   ]
   ++ lib.optional (lib.versionAtLeast qtbase.version "6" && stdenv.isLinux) qtwayland;
 
+  waylandPreExec = ''
+    if [ -n "$WAYLAND_DISPLAY" ]; then
+      export LD_LIBRARY_PATH=${lib.getLib glfw-wayland-minecraft}/lib:"$LD_LIBRARY_PATH"
+    fi
+  '';
+
   postBuild = ''
+    ${lib.optionalString waylandSupport ''
+      qtWrapperArgs+=(--run "$waylandPreExec")
```

Comment:
Why is this separate from the existing `qtWrapperArgs` below?

Reply:
That is covered in previous discussion.
I strongly encourage you to read all of it.